spec/unit/util/execution_spec.rb

```diff
@@ -219,6 +244,36 @@ def stub_process_wait(exitstatus)
           end
         end
 
+        describe "cwd option" do
+          def expect_cwd_to_be(cwd)
+            Puppet::Util::Execution.expects(executor).with(
+              anything,
+              has_entries(:cwd => cwd),
+              anything,
+              anything,
+              anything
+            ).returns(rval)
+          end
+
+          it 'should raise an ArgumentError if the specified working directory does not exist' do
+            cwd = 'cwd'
+            Puppet::FileSystem.expects(:directory?).with(cwd).returns(false)
```

Comment:
this can just be a stub that returns false without `.with(cwd)`

Reply:
Changing `expects` to `stubs` is fine, but we need the `with` to ensure that it only returns false for our cwd instead of some dummy value. Note that the `stubs` is enough to cause this test to pass in case the implementation changes.